### PR TITLE
[Zombienet] test deregister validator

### PR DESCRIPTION
**Pull request by pepoviola**
ping @sandreim 
This add one test to check the correct value of `polkadot_node_is_active_validator`.

Thanks!

### CODE REVIEW

**Comment by sandreim** [comment]
There is an error at startup - `Error: Invalid chain spec raw file generated.` Maybe related to `zombienet:v1.2.42` ?

**Review by sandreim — COMMENTED** [review]
LGTM

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -784,6 +784,32 @@ zombienet-tests-malus-dispute-valid:
   tags:
     - zombienet-polkadot-integration-test
 
+zombienet-tests-deregister-register-validator:
+  stage:                           stage3
+  image:                           "docker.io/paritytech/zombienet:v1.2.42"
+  <<:                              *kubernetes-env
+  <<:                              *zombienet-refs
+  needs:
+    - job:                         publish-polkadot-debug-image
```
Comment (alvicsam) [inline .gitlab-ci.yml:793]:
```suggestion
    - job:                         publish-polkadot-debug-image
      artifacts:                 true
```
It's better to set this variable explicitly

**Review by alvicsam — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -784,6 +784,33 @@ zombienet-tests-malus-dispute-valid:
...
+      artifacts:                 true
```
Comment (alvicsam) [inline .gitlab-ci.yml:794]:
```suggestion
      artifacts:                   true
```
Sorry, wrong indent

**Review by alvicsam — CHANGES_REQUESTED** [review]

**Review by alvicsam — APPROVED** [review]

**Comment by pepoviola** [comment]
> @pepoviola I've fixed an issue in the JS (the stash account was wrong), but the test keeps failing. It seems to be some sort of issue with metric fetching in Zombienet.
> 
> ```
> 1) ( Deregister / Register Validator Smoke )
>        dave: reports polkadot_node_is_active_validator is 0 within 180 secs:
> 
>       AssertionError: expected 1 to equal 0
>       + expected - actual
> 
>       -1
>       +0
> ```
> 
> At the same time I query the respective node metric endpoint and I get:
> 
> ```
> # HELP polkadot_node_is_active_validator Tracks if the validator is in the active set. Updates at session boundary.
> # TYPE polkadot_node_is_active_validator gauge
> polkadot_node_is_active_validator{chain="rococo_local_testnet"} 0
> ```

Ahh thanks for the fix! Checking the metrics cache now.
Thx!

File: zombienet_tests/smoke/0003-deregister-register-validator-smoke.feature [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+Network: ./0003-deregister-register-validator-smoke.toml
+Creds: config
+
+alice: is up
+bob: is up
+charlie: is up
+dave: is up
+
+# ensure is in the validator set
+dave: reports polkadot_node_is_parachain_validator is 1 within 240 secs
```
Comment (sandreim) [inline zombienet_tests/smoke/0003-deregister-register-validator-smoke.feature:11]: We to check both `polkadot_node_is_active_validator` and `polkadot_node_is_parachain_validator`

File: zombienet_tests/smoke/0003-deregister-register-validator-smoke.feature [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+# deregister and check
+alice: js-script ./0003-deregister-register-validator.js with "deregister,dave" return is 0 within 30 secs
+
+# Wait 2 sessions. The authority set change is enacted at curent_session + 2.
+sleep 120 seconds
+dave: reports polkadot_node_is_active_validator is 0 within 180 secs
```
Comment (sandreim) [inline zombienet_tests/smoke/0003-deregister-register-validator-smoke.feature:18]: Also here.

**Comment by pepoviola** [comment]
ping @sandreim `deregister/register` test fixed.
Thanks!

File: zombienet_tests/smoke/0003-deregister-register-validator-smoke.feature (hunk 4 above)
Comment (pepoviola) [reply zombienet_tests/smoke/0003-deregister-register-validator-smoke.feature:20]: done :)

File: zombienet_tests/smoke/0003-deregister-register-validator-smoke.toml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  name = "bob"
+  args = [ "-lruntime=debug,parachain=trace" ]
+
+  [[relaychain.nodes]]
+  name = "charlie"
+  args = [ "-lruntime=debug,parachain=trace" ]
+
+  [[relaychain.nodes]]
+  name = "dave"
+  args = [ "-lruntime=debug,parachain=trace" ]
```
Comment (chevdor) [inline zombienet_tests/smoke/0003-deregister-register-validator-smoke.toml:23]:
```suggestion
  args = [ "-lruntime=debug,parachain=trace" ]
  
```

File: zombienet_tests/smoke/0003-deregister-register-validator.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+              unsub();
+              return reject();
+            }
+          });
+    });
+
+    return 0;
+}
+
+module.exports = { run }
```
Comment (chevdor) [inline zombienet_tests/smoke/0003-deregister-register-validator.js:46]:
```suggestion
module.exports = { run }

```

**Review by chevdor — APPROVED** [review]
LGTM besides the 2 missing EOF

**Review by sandreim — APPROVED** [review]

**Comment by sandreim** [comment]
bot merge

**Pull request merged**